### PR TITLE
Adding SMB enhacements

### PR DESCRIPTION
**Pull request by Bhavneet-Sharma**
# Description
Adding the Support for run as root, along with other boolean flags like allow_delete_readonly, allow_execute_always and inheritable_path_acl

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|#63 |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [X] I have performed Ansible Sanity test using --docker default
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit Testing
- [x] Functional Testing

### CODE REVIEW

File: tests/unit/requirements.txt [hunk 1]
```diff
@@ -4,4 +4,4 @@ mock
 pytest-mock
 pytest-cov
 pytest-forked
-coverage==4.5.4
\ No newline at end of file
+coverage
```
Comment (trisha-dell) [inline tests/unit/requirements.txt:7]: Add new line at the end

File: plugins/modules/smb.py [hunk 2]
```diff
@@ -422,48 +494,70 @@
...
-            description: Path of the SMB Share
+            description: Path of the SMB Share.
             type: str
         permission:
-            description: permission on the of the SMB Share for user/group/wellknown
+            description: permission on the of the SMB Share for user/group/wellknown/
             type: list
         file_create_mask:
-            description: File create mask bit for SMB Share
+            description: File create mask bit for SMB Share/
```
Comment (trisha-dell) [inline plugins/modules/smb.py:519]: Replace / with .

File: tests/unit/requirements.txt (hunk 1 above)
Comment (Bhavneet-Sharma) [reply tests/unit/requirements.txt]: Fixed

File: plugins/modules/smb.py (hunk 2 above)
Comment (Bhavneet-Sharma) [reply plugins/modules/smb.py]: Fixed

**Review by trisha-dell — DISMISSED** [review]

**Review by sachin-apa — DISMISSED** [review]

**Review by felixs88 — APPROVED** [review]

**Review by trisha-dell — APPROVED** [review]

**Review by trisha-dell — DISMISSED** [review]

**Review by trisha-dell — APPROVED** [review]

**Pull request merged**